data/transactions/logic/eval.go

```diff
@@ -4875,6 +4887,11 @@ func (cx *EvalContext) resolveAsset(ref uint64) (aid basics.AssetIndex, err erro
 	if ref < uint64(len(cx.txn.Txn.ForeignAssets)) {
 		return basics.AssetIndex(cx.txn.Txn.ForeignAssets[ref]), nil
 	}
+
+	if cx.UnlimitedResourceAccess {
```

Comment:
I think I would have expected this check to go into `cx.availableApp`, but I think I see why you kept it out.

Was your thinking that we wouldn't want `cx.AvailableAsset(2)` returning true, if the code was actually trying to use slot 2? That is, you wanted lines 4887-4889 to get a chance?

If that is the sole reason, I think I would advocate going ahead with moving the `cx.UnlimitedResourceAccess` check into `cx.availableAsset` but not allowing the unlimit flag to work for assets under 256.  It feels more natural to say that the flag makes things available, than to add the checks afterward, and by not working on assets under 256, we avoid the slot ambiguity.

And we don't need the unlimit flag to provide access to low number resources anyway, since they are illegal.

Reply:
This makes sense, I'll take this suggestion